card/src/main/java/com/adyen/checkout/card/internal/ui/state/StoredCardComponentStateExt.kt

```diff
@@ -0,0 +1,114 @@
+
+import com.adyen.checkout.card.internal.ui.model.CardComponentParams
+import com.adyen.checkout.core.components.data.PaymentComponentData
+import com.adyen.checkout.core.components.paymentmethod.CardPaymentMethod
+import com.adyen.checkout.core.old.internal.util.runCompileOnly
+import com.adyen.checkout.cse.EncryptedCard
+import com.adyen.checkout.cse.EncryptionException
+import com.adyen.checkout.cse.UnencryptedCard
+import com.adyen.checkout.cse.internal.BaseCardEncryptor
+import com.adyen.threeds2.ThreeDS2Service
+
+@Suppress("ReturnCount", "LongParameterList")
+internal fun StoredCardComponentState.toPaymentComponentState(
+    componentParams: CardComponentParams,
+    cardEncryptor: BaseCardEncryptor,
+    checkoutAttemptId: String,
+    storedPaymentMethodId: String?,
+    onEncryptionFailed: (EncryptionException) -> Unit,
+    onPublicKeyNotFound: (RuntimeException) -> Unit,
+): CardPaymentComponentState {
+    val publicKey = componentParams.publicKey(onPublicKeyNotFound = onPublicKeyNotFound)
+        ?: return invalidCardPaymentComponentState()
+
+    val encryptedCard = encryptCard(
+        cardEncryptor = cardEncryptor,
+        publicKey = publicKey,
+        onEncryptionFailed = onEncryptionFailed,
+    ) ?: return invalidCardPaymentComponentState()
+
+    val cardPaymentMethod = createPaymentMethod(
+        storedCardId = storedPaymentMethodId,
+        encryptedCard = encryptedCard,
+        checkoutAttemptId = checkoutAttemptId,
+    )
+
+    val paymentComponentData = createPaymentComponentData(cardPaymentMethod, componentParams)
+
+    return createPaymentComponentState(paymentComponentData)
+}
+
+@Suppress("LongParameterList")
+private fun StoredCardComponentState.encryptCard(
+    cardEncryptor: BaseCardEncryptor,
+    publicKey: String,
+    onEncryptionFailed: (EncryptionException) -> Unit,
+): EncryptedCard? {
+    val unencryptedCardBuilder = UnencryptedCard.Builder()
+    return try {
+        // TODO - Card. Add isCvcHidden check
+        val cvc = securityCode.text
+        if (cvc.isNotEmpty()) unencryptedCardBuilder.setCvc(cvc)
+        cardEncryptor.encryptFields(unencryptedCardBuilder.build(), publicKey)
+    } catch (e: EncryptionException) {
+        onEncryptionFailed(e)
+        null
+    }
+}
+
+private fun invalidCardPaymentComponentState() = CardPaymentComponentState(
+    data = PaymentComponentData(null, null, null),
+    isValid = false,
+)
+
+private fun createPaymentComponentData(
+    cardPaymentMethod: CardPaymentMethod,
+    componentParams: CardComponentParams
+) = PaymentComponentData(
+    paymentMethod = cardPaymentMethod,
+    storePaymentMethod = null,
+    shopperReference = componentParams.shopperReference,
+    order = null,
+    amount = componentParams.amount,
+)
+
+private fun createPaymentComponentState(
+    paymentComponentData: PaymentComponentData<CardPaymentMethod>,
+): CardPaymentComponentState {
+    return CardPaymentComponentState(
```

Comment:
Nits: This could also be:
```suggestion
) = CardPaymentComponentState(
```